cdap-api/src/main/java/co/cask/cdap/api/data/batch/PartitionedFileSetInputContext.java

```diff
@@ -19,13 +19,22 @@
 import co.cask.cdap.api.dataset.lib.PartitionKey;
 import co.cask.cdap.api.dataset.lib.PartitionedFileSet;
 
+import java.util.List;
+
 /**
  * Exposes information about the {@link PartitionedFileSet} input configured for this task.
  */
 public interface PartitionedFileSetInputContext extends InputContext {
 
   /**
    * Returns the {@link PartitionKey} of the input configured for this task.
+   * In case of CombineFileInputFormat, this will be the PartitionKey currently being processed by the task.
    */
   PartitionKey getInputPartitionKey();
+
+  /**
+   * Returns a list of {@link PartitionKey}s of the input configured for this task. There can be multiple PartitionKeys
+   * for a single task if using CombineFileInputFormat.
+   */
+  List<PartitionKey> getInputPartitionKeys();
```

Comment:
I thought it may be useful to also expose what is the entire list of partition keys that are being processed.